src/components/PeopleList.tsx

```diff
@@ -0,0 +1,223 @@
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { Person } from '../types';
+import { PeopleItem } from './PeopleItem';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[],
+};
+
+export const PeopleList: React.FC<Props> = ({
+  people,
+}) => {
+  const [sortPeople, setSortPeople] = useState<Person[]>(people);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+
+  const getSearchSort = useCallback((value: string) => {
+    if (sort !== value) {
+      return { sort: value, order: null };
+    }
+
+    return (!order)
+      ? { sort: value, order: 'desc' }
+      : { sort: null, order: null };
+  }, [sort, order]);
+
+  const handlerSortPeople = useCallback((
+    sortParam: 'name' | 'sex' | 'born' | 'died',
+  ) => {
+    return [...people].sort((a, b) => {
+      const personA = a[sortParam];
+      const personB = b[sortParam];
+
+      if (typeof personA === 'string' && typeof personB === 'string') {
+        return personA.localeCompare(personB);
+      }
+
+      return (+personA) - (+personB);
+    });
+  }, [people]);
+
+  const getSortedPeople = useCallback(() => {
+    let sortedPeople = [...people];
+
+    if (sort) {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+        case 'born':
+        case 'died':
+          sortedPeople = handlerSortPeople(sort);
+          break;
+
+        default:
+          sortedPeople = people;
+      }
+
+      return order ? sortedPeople.reverse() : sortedPeople;
+    }
+
+    return sortedPeople;
+  }, [sort, people, order]);
+
+  const getFilteredPeople = useCallback((sortedPeoples: Person[]) => {
+    return sortedPeoples
+      .filter(person => {
+        const normalizeQuery = query.trim().toLocaleLowerCase();
+        const { name, motherName, fatherName } = person;
+
+        if (normalizeQuery) {
+          if (!!motherName
+            && motherName.toLocaleLowerCase().includes(normalizeQuery)
+          ) {
+            return true;
+          }
+
+          if (!!fatherName
+            && fatherName.toLocaleLowerCase().includes(normalizeQuery)
+          ) {
+            return true;
+          }
+
+          return name.toLocaleLowerCase().includes(normalizeQuery);
+        }
+
+        return true;
+      })
+      .filter(person => {
+        if (sex) {
+          return person.sex === sex;
+        }
+
+        return true;
+      })
+      .filter(person => {
+        if (!!centuries && !!centuries.length) {
+          const { born } = person;
+          const bornCentury = Math.ceil(+born / 100);
+
+          return centuries.includes(bornCentury.toString());
+        }
+
+        return true;
+      });
```

Comment:
you are using a lot of filters methods here, let's improve your code
let's try to combine this filters to one filter with the all conditions